lesson1/exercise4.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Exercise 4: What is all this stuff"
+nav_order: 5
+parent: 1 - First Steps with Java
+---
+
+# Exercise 4: What is all this stuff
+
+We typed or copy-pasted bits of Java code without fully understanding what certain words or certain brackets mean. Here we drop the curtain and show the meaning behind that text.
+
+## Objective
+
+In this exercise we will learn about:
+ * Different kinds of brackets
+ * Keywords
+ * Importance of structure
+
+At the end of this you should understand:
+ * How do we structure code using the curly brackets
+ * How and where are round brackets used
+ * What are keywords and provide examples of some Java keywords
+ * What is and why do we structure programs
+
+## Brackets
+
+Programming has a deep connection with Mathematics, first computing devices were built to solve simple math problems such as counting items or simple arithmetic. That's why there's a great deal of brackets that are in use -- they come from math.
+
+**INFO:** If you're interested in more on this, look up the article on the [History of computing hardware](https://en.wikipedia.org/wiki/History_of_computing_hardware) on Wikipedia, and this [question and answer](https://softwareengineering.stackexchange.com/questions/136987/what-does-mathematics-have-to-do-with-programming) on StackExchange.
```

Comment:
What about line breaks?

Reply:
I've added a bit in the "Other ways of structuring code" section